File: 0001-two-sum/0001-two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var ch = Map();
    var ans = [-1 , -1];
    for(int i = 0; i < nums.length; i++){
        if(ch.containsKey(target - nums[i])) {
            ans[0] = i;
            ans[1] = ch[target - nums[i]];
            break;
        }

        ch[nums[i]] = i;
    }

    return ans;
  }
}

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
   mp := make(map[int]int)
    ans := []int{0 , 0};

    for i := range len(nums) {
        if val,ok := mp[target - nums[i]]; ok {
            ans[0] = i;
            ans[1] = val;
            break;
        }
        mp[nums[i]] = i;
    }

    return ans;
}